packages/core/src/components/toast/toast2.tsx

```diff
@@ -44,7 +44,8 @@ export const Toast2 = React.forwardRef<HTMLDivElement, ToastProps>((props, ref)
     const clearTimeout = React.useCallback(() => setIsTimeoutStarted(false), []);
 
     // Per docs: "Providing a value less than or equal to 0 will disable the timeout (this is discouraged)."
-    const isTimeoutEnabled = timeout != null && timeout > 0;
+    // Per github, issue 6742: "timeout: Infinity should also behave the same as timeout: 0"
```

Comment:
I think we can remove both these comments, they don't add anything that the code does not provide here.

Reply:
Also, lets update the docs for the timeout prop to mention the behavior of `Infinity`.